pkg/acsengine/defaults_test.go

```diff
@@ -497,6 +497,108 @@ func TestStorageProfile(t *testing.T) {
+		t.Fatalf("Master VMSS, AzureCNI: MasterProfile AvailabilityProfile did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.AvailabilityProfile, api.VirtualMachineScaleSets)
+	}
+	if properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet != DefaultKubernetesSubnet {
+		t.Fatalf("Master VMSS, AzureCNI: MasterProfile ClusterSubnet did not have the expected default configuration, got %s, expected %s",
+			properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet, DefaultKubernetesSubnet)
+	}
+	if properties.MasterProfile.FirstConsecutiveStaticIP != DefaultFirstConsecutiveKubernetesStaticIPVMSS {
+		t.Fatalf("Master VMSS, AzureCNI: MasterProfile FirstConsecutiveStaticIP did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.FirstConsecutiveStaticIP, DefaultFirstConsecutiveKubernetesStaticIPVMSS)
+	}
+	if properties.MasterProfile.Subnet != DefaultKubernetesMasterSubnet {
+		t.Fatalf("Master VMSS, AzureCNI: MasterProfile Subnet did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.Subnet, DefaultKubernetesMasterSubnet)
+	}
+	if properties.MasterProfile.AgentSubnet != DefaultKubernetesAgentSubnet {
+		t.Fatalf("Master VMSS, AzureCNI: MasterProfile AgentSubnet did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.AgentSubnet, DefaultKubernetesAgentSubnet)
+	}
+
+	properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet = ""
+	properties.OrchestratorProfile.KubernetesConfig.NetworkPlugin = "kubenet"
+	setPropertiesDefaults(&mockCS, false, true)
+	if properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet != DefaultKubernetesClusterSubnet {
+		t.Fatalf("Master VMSS, kubenet: MasterProfile ClusterSubnet did not have the expected default configuration, got %s, expected %s",
+			properties.OrchestratorProfile.KubernetesConfig.ClusterSubnet, DefaultKubernetesClusterSubnet)
+	}
+	if properties.MasterProfile.Subnet != DefaultKubernetesMasterSubnet {
+		t.Fatalf("Master VMSS, kubenet: MasterProfile Subnet did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.Subnet, DefaultKubernetesMasterSubnet)
+	}
+	if properties.MasterProfile.FirstConsecutiveStaticIP != DefaultFirstConsecutiveKubernetesStaticIPVMSS {
+		t.Fatalf("Master VMSS, kubenet: MasterProfile FirstConsecutiveStaticIP did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.FirstConsecutiveStaticIP, DefaultFirstConsecutiveKubernetesStaticIPVMSS)
+	}
+	if properties.MasterProfile.AgentSubnet != DefaultKubernetesAgentSubnet {
+		t.Fatalf("Master VMSS, kubenet: MasterProfile AgentSubnet did not have the expected default configuration, got %s, expected %s",
+			properties.MasterProfile.AgentSubnet, DefaultKubernetesAgentSubnet)
+	}
+	properties.MasterProfile.AvailabilityProfile = api.AvailabilitySet
```

Comment:
ditto to comment above: let's make it clear we are testing something different starting here, new test or reset mock object + comments